File: September 2024/September 23/games.py
def how_many_games(p, d, m, s):
    games = 0
    funds = s
    price = p

    while funds > 0:
        funds -= price
        if price-d < m:
            price = m
        else:
            price -= d
        if funds < 0:
            break
        else:
            games+=1
    return games

print(how_many_games(20, 3, 6, 85))

File: September 2024/September 23/games.js
function howManyGames(p, d, m, s) {
    let games = 0
    let funds = s
    let price = p
  
    while(funds > 0){
      funds-= price
      if(price - d < m){
        price = m 
      }else{
        price-=d
      }
      if(funds < 0){
        break
      }else{
        games++
      }
      console.log(funds)
    }
    
    return games
  }